features/admin.roles.v2/components/wizard-updated/role-basics.tsx

```diff
@@ -118,6 +119,15 @@ export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicPr
     const { organizationType } = useGetCurrentOrganizationType();
     const isSubOrg: boolean = organizationType === OrganizationType.SUBORGANIZATION;
 
+    // Filter out readonly applications and shared parent applications.
+    const filteredApplicationList: ApplicationListItemInterface[] = useMemo(() => {
+        return applicationList?.applications
+            ?.filter((application: ApplicationListItemInterface) =>
+                !RoleConstants.READONLY_APPLICATIONS_CLIENT_IDS.includes(application?.clientId) &&
```

Comment:
We wouldn't require READONLY_APPLICATIONS_CLIENT_IDS filtering if we pass `excludeSystemPortals` param to useApplicationList hook in L109